### PR TITLE
Add senior delegate to automated results reminder

### LINKED ISSUE #2089: Add senior delegate to automated results reminder
I just learned that senior delegates are CC'd for late report reminders, but not for late results reminders. We should CC seniors for the latter as well.

### PR DESCRIPTION
**Pull request by SqAtx**
Fixes #2089 

We fixed a thing or two along the way -> negative lines of code 8)

### CODE REVIEW

File: WcaOnRails/app/mailers/competitions_mailer.rb [hunk 1]
```diff
@@ -59,9 +59,13 @@ def submit_report_nag(competition)
...
       reply_to: "board@worldcubeassociation.org",
       subject: "#{competition.name} Delegate Report",
     )
   end
-end
+
+  private def delegates_to_senior_delegates_email(delegates)
+    delegates.map { |delegate| delegate.senior_delegate&.email }.uniq.compact
+  end
+end
```
Comment (jfly) [inline WcaOnRails/app/mailers/competitions_mailer.rb]: Missing a newline here. That's why travis is failing (see https://travis-ci.org/thewca/worldcubeassociation.org/builds/292413212#L860)

Comment (SqAtx) [reply WcaOnRails/app/mailers/competitions_mailer.rb]: Oops my bad, I wasn't sure. Now I know ^^

**Comment by SqAtx** [comment]
Oops.. Now Travis is happy but it looks like code coverage decreased somehow

**Comment by jfly** [comment]
Thanks! Don't worry about the code coverage. I've merged and deployed.

**Pull request merged**